src/main/java/uk/gov/companieshouse/api/accounts/validation/CreditorsAfterOneYearValidator.java

```diff
@@ -0,0 +1,340 @@
+    }
+
+    private void checkIfPreviousBalanceAndNoteValuesAreEqual(Errors errors, CreditorsAfterOneYear creditorsAfterOneYear,
+                                                             BalanceSheet previousPeriodBalanceSheet) {
+
+        if (!isPreviousPeriodBalanceSheetCreditorsAfterOneYearNull(previousPeriodBalanceSheet) &&
+            (!isCreditorsAfterOneYearNotePreviousTotalNull(creditorsAfterOneYear)) &&
+            (!creditorsAfterOneYear.getPreviousPeriod().getTotal().equals(
+                previousPeriodBalanceSheet.getOtherLiabilitiesOrAssets().getCreditorsAfterOneYear()))) {
+
+            addError(errors, currentBalanceSheetNotEqual, CREDITORS_AFTER_PREVIOUS_PERIOD_TOTAL_PATH);
+        }
+    }
+
+    private void checkIsPreviousBalanceNullAndNoteNot(Errors errors, CreditorsAfterOneYear creditorsAfterOneYear,
+                                                      BalanceSheet previousPeriodBalanceSheet) {
+
+        if (isPreviousPeriodBalanceSheetCreditorsAfterOneYearNull(previousPeriodBalanceSheet) &&
+            (!isCreditorsAfterOneYearNotePreviousTotalNull(creditorsAfterOneYear))) {
+
+            addError(errors, previousBalanceSheetNotEqual, CREDITORS_AFTER_PREVIOUS_PERIOD_TOTAL_PATH);
+        }
+    }
+
+    private void checkIfPreviousNoteIsNullAndBalanceNot(Errors errors, CreditorsAfterOneYear creditorsAfterOneYear,
+                                                        BalanceSheet previousPeriodBalanceSheet) {
+
+        if (isCreditorsAfterOneYearNotePreviousTotalNull(creditorsAfterOneYear) &&
+            (!isPreviousPeriodBalanceSheetCreditorsAfterOneYearNull(previousPeriodBalanceSheet))) {
+
+            addError(errors, previousBalanceSheetNotEqual, CREDITORS_AFTER_PREVIOUS_PERIOD_TOTAL_PATH);
+        }
+    }
+
+    private boolean isCreditorsAfterOneYearNotePreviousTotalNull(CreditorsAfterOneYear creditorsAfterOneYear) {
+        return creditorsAfterOneYear.getPreviousPeriod() == null ||
+            creditorsAfterOneYear.getPreviousPeriod().getTotal() == null;
+    }
+
+    private boolean isPreviousPeriodBalanceSheetCreditorsAfterOneYearNull(BalanceSheet previousPeriodBalanceSheet) {
+        return previousPeriodBalanceSheet == null ||
+            previousPeriodBalanceSheet.getOtherLiabilitiesOrAssets() == null ||
+            previousPeriodBalanceSheet.getOtherLiabilitiesOrAssets().getCreditorsAfterOneYear() == null;
+    }
+
+    private BalanceSheet getCurrentPeriodBalanceSheet(HttpServletRequest request,
+                                                      String companyAccountsId) throws DataException {
+
+        String currentPeriodId = currentPeriodService.generateID(companyAccountsId);
+
+        ResponseObject<uk.gov.companieshouse.api.accounts.model.rest.CurrentPeriod> currentPeriodResponseObject;
+
+        try {
+            currentPeriodResponseObject = currentPeriodService.findById(currentPeriodId, request);
+        } catch (MongoException me) {
```

Comment:
`CurrentPeriodService#findById` should throw a `DataException`, so there's no need to catch a `MongoException`

Reply:
altered